File: Individual Presentation - RISC Processors/benchmark/integral/integral.c
#include <stdio.h>
#include <math.h>
#include <time.h>

int main(){
  double sum = 0, delta = 0.025, x = -5;
  double y, z;
  int i, j,k;
  clock_t t_start, t_stop;
  
  printf("-----\n* C *\n-----\n");
  printf("Approximation of a triple integral over a ball.\n");
  
  t_start = clock();
 
  for(i=0; i <400; i++)
   {x += delta;
   y = -5;
   for(j=0; j<400; j++)
     {y += delta;
     z=-5;
     for(k=0;k<400;k++)
       {z += delta;
       if(x*x+y*y+z*z<25)
	 sum += exp(-x*x)*cos(y*y)*cos(z*z);}}}
  sum=sum*delta*delta*delta;
  t_stop = clock();
 printf("integral = %f10\n", sum);
 printf("elapsed time: %f10 (s)\n", ((double) t_stop - t_start) / CLOCKS_PER_SEC);
 return 0;
}

File: Individual Presentation - RISC Processors/benchmark/integral/integral.f
* Integral benchmark adapted from https://cboard.cprogramming.com/c-programming/47759-mac-os-x-users-c-programmers.html
      PROGRAM integral
* Declarations
	  REAL isum, delta, x, y, z, t_start, t_stop
	  INTEGER i, j, k
	  isum = 0.0
	  delta = 0.025
	  x = -5.0
	  
	  WRITE (*,*) '-----------'
	  WRITE (*,*) '* FORTRAN *'
	  WRITE (*,*) '-----------'
	  WRITE (*,*) 'Approximation of a triple integral over a ball.'
	  
* Need to begin clock here.
      call CPU_TIME(t_start)

      DO 10 i=0,400
	     x = x + delta
		 y = -5.0
		 DO 15 j=0,400
		    y = y + delta
			z = -5.0
			DO 20 k=0,400
			   z = z + delta
			   IF ((x*x+y*y+z*z).LT.25.0) THEN
			      isum = isum + EXP(-x*x)*COS(y*y)*COS(z*z)
			   ENDIF
   20		CONTINUE
   15	 CONTINUE
   10 CONTINUE
      isum = isum*delta*delta*delta
* print execution time.
      call CPU_TIME(t_stop)
	  WRITE (*,*) 'integral = ', isum
	  WRITE (*,*) 'elapsed time: ', t_stop - t_start, ' (s)'	  
      END	  